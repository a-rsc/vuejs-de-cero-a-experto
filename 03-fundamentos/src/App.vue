<template>
  <div>
    <!-- <img alt="Vue logo" src="./assets/logo.png"> -->
    <!-- <CounterVue title="1r componente" atributo="15" v-bind:start="15" />
    <CounterVue /> -->
    <IndecisionVue />
  </div>
</template>

<script>
// import HelloWorld from './components/HelloWorld.vue'
// import HelloWorld from '@/components/HelloWorld.vue'
import CounterVue from '@/components/Counter.vue'
import IndecisionVue from '@/components/Indecision.vue'

export default {
  name: 'App',
  components: {
    CounterVue,
    IndecisionVue
  }
}
</script>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
