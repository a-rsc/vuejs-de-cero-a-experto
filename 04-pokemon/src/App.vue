<template>
  <Page></Page>
</template>

<script>
/* Importaci√≥n tradicional, pero cuando se construye forma parte del bundle. Se puede hacer de otras maneras. */
import Page from '@/components/PokemonPage' // src

export default {
  name: 'App',
  components: {
    Page
  }
}
</script>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
