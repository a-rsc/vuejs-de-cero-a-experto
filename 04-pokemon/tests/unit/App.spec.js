// npm run test:unit indecision
// Ha aparecido una carpeta __snapshots___
// Si cambiamos se producirá un error

// Si el cambio es nuevo -> npm run test:unit -u indecision

// mount -> monta todo el componente de pruebas
// shallowMount -> parecido pero menos algunas propiedades y funciones...
// import { mount } from '@vue/test-utils'
import { shallowMount } from '@vue/test-utils'
import App from '@/App'

describe('App Component', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallowMount(App) // aplicación que corre independientemente
    })

    // la función it es muy parecida
    test('Debe de hacer match con el snapshot', () => {
        expect( wrapper.html() ).toMatchSnapshot()
    })
})
