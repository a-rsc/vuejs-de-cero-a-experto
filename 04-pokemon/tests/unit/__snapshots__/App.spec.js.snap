// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Debe de hacer match con el snapshot 1`] = `<page-stub></page-stub>`;
